components/eam/src/chemistry/mozart/mo_extfrc.F90

```diff
@@ -339,9 +351,27 @@ subroutine extfrc_set( lchnk, zint, frcing, ncol )
        call outfld( xfcname, frcing(:ncol,:,n), ncol, lchnk )
 
        frcing_col(:ncol) = 0._r8
+       frcing_tmp(:ncol,:) = 0._r8
+       if (trim(forcings(m)%species) == 'NO2' .and. history_gaschmbudget_2D) then
+            no2_tdacf(:ncol,:) = 0._r8
+       endif
        do k = 1,pver
-          frcing_col(:ncol) = frcing_col(:ncol) + frcing(:ncol,k,n)*(zint(:ncol,k)-zint(:ncol,k+1))*km_to_cm
+          frcing_tmp(:ncol,k) = frcing(:ncol,k,n)*(zint(:ncol,k)-zint(:ncol,k+1))*km_to_cm
+          frcing_col(:ncol) = frcing_col(:ncol) + frcing_tmp(:ncol,k)
+          !frcing_col(:ncol) = frcing_col(:ncol) + frcing(:ncol,k,n)*(zint(:ncol,k)-zint(:ncol,k+1))*km_to_cm
+
+          if (trim(forcings(m)%species) == 'NO2' .and. history_gaschmbudget_2D) then
+          !kgn per m2 per second: * 14.00674 * 1.65979e-24 * 1.e-3 * 1.e4
+          !             nb: 1.65979e-24 = 1/avo
+               no2_tdacf(:ncol,k) = frcing_tmp(:ncol,k) * 14.00674_r8 * 1.65979e-24_r8 * 10._r8 
```

Comment:
The avogadro constant is defined in mo_constants. It would be better to avoid hard-coding it by:
use mo_constants, only : avogadro